homeassistant/components/nexia/climate.py

```diff
@@ -0,0 +1,524 @@
+        if new_heat_temp and new_heat_temp + deadband > max_temp:
+            new_heat_temp = max_temp - deadband
+        if new_cool_temp and new_cool_temp - deadband < min_temp:
+            new_cool_temp = min_temp + deadband
+
+        # Check that we're within the deadband range, fix it if we're not
+        if new_heat_temp and new_heat_temp != cur_heat_temp:
+            if new_cool_temp - new_heat_temp < deadband:
+                new_cool_temp = new_heat_temp + deadband
+        if new_cool_temp and new_cool_temp != cur_cool_temp:
+            if new_cool_temp - new_heat_temp < deadband:
+                new_heat_temp = new_cool_temp - deadband
+
+        self._device.set_heat_cool_temp(
+            heat_temperature=new_heat_temp,
+            cool_temperature=new_cool_temp,
+            set_temperature=set_temp,
+        )
+        self.schedule_update_ha_state()
+
+    @property
+    def is_aux_heat(self):
+        """Emergency heat state."""
+        return "on" if self._device.thermostat.is_emergency_heat_active() else "off"
+
+    @property
+    def device_info(self):
+        """Return the device_info of the device."""
+        return {
+            "identifiers": {(DOMAIN, self._device.zone_id)},
+            "name": self._device.get_name(),
+            "model": self._device.thermostat.get_model(),
+            "sw_version": self._device.thermostat.get_firmware(),
+            "manufacturer": MANUFACTURER,
+            "via_device": (DOMAIN, self._device.thermostat.thermostat_id),
+        }
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+
+        (min_temp, max_temp) = self._device.thermostat.get_setpoint_limits()
+        data = {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+            ATTR_FAN_MODE: self._device.thermostat.get_fan_mode(),
+            ATTR_HVAC_MODE: self.mode,
+            ATTR_TARGET_TEMP_HIGH: self._device.get_cooling_setpoint(),
+            ATTR_TARGET_TEMP_LOW: self._device.get_heating_setpoint(),
+            ATTR_TARGET_TEMP_STEP: 1.0
+            if self._device.thermostat.get_unit() == UNIT_FAHRENHEIT
+            else 0.5,
+            ATTR_MIN_TEMP: min_temp,
+            ATTR_MAX_TEMP: max_temp,
+            ATTR_FAN_MODES: FAN_MODES,
+            ATTR_HVAC_MODES: self.hvac_modes,
+            ATTR_PRESET_MODE: self._device.get_preset(),
+            ATTR_HOLD_MODES: self._device.get_presets(),
+            ATTR_SETPOINT_STATUS: self._device.get_setpoint_status(),
+            ATTR_ZONE_STATUS: self._device.get_status(),
+            ATTR_THERMOSTAT_ID: self._device.thermostat.thermostat_id,
+            ATTR_ZONE_ID: self._device.zone_id,
+        }
+
+        if self._device.thermostat.has_emergency_heat():
+            data.update(
+                {
+                    ATTR_AUX_HEAT: "on"
+                    if self._device.thermostat.is_emergency_heat_active()
+                    else "off"
```

Comment:
We should avoid using ternary statements if they span multiple lines.

Reply:
fixed